Creating-Objects.md

```diff
@@ -1,12 +1,105 @@
-       * `description`: A string describing the object.
-       * (optional) `edit_html_filename`: The name of the file in (3) below, if it exists.
-       * (optional) `edit_js_filename`: The name of the file in (4) below, if it exists.
-       * `default_value`: This is needed for any object that is used as an input parameter to a rule. It represents the default value of that parameter.
-       * `normalize`: A function that is given an instance of the object in question and is responsible for validating and normalizing it. The supplied instance is a native Python object.
-  1. In `extensions/objects/models/objects_test.py` add new tests to the `ObjectNormalizationUnitTests` class; these will be run automatically with the other tests.
-  1. (optional) In `extensions/objects/templates` add a file `my_object_editor.html` specifying an interface in which objects of this type can be submitted. This is only necessary if editors will be entering parameters that use this object type.
-  1. (optional) In `extensions/objects/templates`, create `myObjectEditor.js`, the JavaScript directive associated with `my_object_editor.html`. You should base this (and the HTML file) on one of the pre-existing ones.
+## Table of Contents
+
+* [Introduction](#introduction)
+* [How objects are defined](#how-objects-are-defined)
+  * [Python classes](#python-classes)
+  * [Angular components](#angular-components)
+* [Create a new object](#create-a-new-object)
+* [Using objects](#using-objects)
+
+## Introduction
+
+When collecting user input, we often define the kind of input we expect using [[schemas|Schemas]]. These schemas can describe arbitrarily nested collections of base types we call _objects_. For example, consider this schema:
+
+```json
+{
+  "type": "dict",
+  "properties": [
+    {
+      "name": "version",
+      "schema": {
+        "type": "int"
+      }
+    }
+  ]
+}
+```
+
+The schema would match a dictionary like this:
+
+```json
+{"version": 34}
+```
+
+The `dict` type is a collection, but the `int` type is an object. We define these objects as classes in [`extensions/objects/models/objects.py`](https://github.com/oppia/oppia/blob/develop/extensions/objects/models/objects.py) and components in [`extensions.objects/templates`](https://github.com/oppia/oppia/blob/develop/extensions/objects/templates).
+
+## How objects are defined
+
+### Python classes
+
+The Python classes for objects are defined in `objects.py`. For example, here's the class for the `int` type:
+
+```python
+class Int(BaseObject):
+    """Integer class."""
+
+    description = 'An integer.'
+    default_value = 0
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        return {
+            'type': 'int'
+        }
+```
+
+Object classes all meet the following criteria:
+
+* Inherit from either `BaseObject` or another object class.
```

Comment:
"They inherit...", "Their `description` class variable is set to...", etc.

That way, you describe them in terms of what they are (as suggested by the above intro to the list) and not in terms of what the developer does.

Reply:
done